decidim-core/app/controllers/decidim/authorizations_controller.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+require_dependency "decidim/application_controller"
+
+module Decidim
+  # This controller allows users to create and destroy their authorizations. It
+  # shouldn't be necessary to expand it to add new authorization schemes.
+  class AuthorizationsController < ApplicationController
+    helper_method :handler, :handlers
+    before_action :valid_handler, only: [:new, :create]
+    before_action :only_one_handler?, only: [:index]
+
+    def create
+      AuthorizeUser.call(handler) do
+        on(:ok) do
+          flash[:notice] = t("authorizations.create.success", scope: "decidim")
+          redirect_to account_path
+        end
+
+        on(:invalid) do
+          flash[:alert] = t("authorizations.create.error", scope: "decidim")
+          render action: :new
+        end
+      end
+    end
+
+    def destroy
+      @authorization = current_user.authorizations.find(params[:id])
+      @authorization.destroy
+      flash[:notice] = t("authorizations.destroy.success", scope: "decidim")
+      redirect_to account_path
+    end
+
+    def handler
+      @handler ||= AuthorizationHandler.handler_for(handler_name, handler_params)
+    end
+
+    def handlers
+      @handlers ||= Decidim.authorization_handlers
+    end
+
+    protected
+
+    def handler_params
+      (params[:authorization_handler] || {}).merge(user: current_user)
+    end
+
+    def handler_name
+      params[:handler] || params[:authorization_handler][:handler_name]
+    end
+
+    def valid_handler
+      return true if handler
+
+      logger.warn "Invalid authorization handler given: #{handler_name} doesn't exist, you haven't added it to" \
```

Comment:
Liar! You're not checking if it's a descendant of `Decidim::AuthorizationHandler` (which is ok for me, but the message is misleading).